dataprofiler/profilers/profiler_options.py

```diff
@@ -6,11 +6,16 @@
 import copy
```

Comment:
Consists of: 

- `load_to_dict()`
- creating `Generic` typing for `BaseOptions`
- updating the typing for some attributes